### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by Dragos0000**

### CODE REVIEW

File: src/reasoning-layer-lib/connectors-reasoning-layer.xsl [hunk 1]
```diff
@@ -103,24 +103,28 @@
         <xsl:variable name="targetRole" select="f:lexicalQNameToWords($connectorName)"/>
         <xsl:variable name="targetRoleURI"
             select="f:buildURIfromLexicalQName($targetRole, fn:false())"/>
-        <xsl:choose>
-            <xsl:when test="fn:count($connectorsWithSameName) = 1">
+<!--        This will filter out all Dependencies that are from a Class to an Enumeration-->
```
Comment (costezki) [inline src/reasoning-layer-lib/connectors-reasoning-layer.xsl:106]: well done commenting what these conditions do

File: src/reasoning-layer-lib/connectors-reasoning-layer.xsl [hunk 2]
```diff
@@ -103,24 +103,28 @@
...
+        <xsl:variable name="filteredConnectorsWithSameName"
+            select="$connectorsWithSameName[not(./properties/@ea_type = 'Dependency' and ./source/model/@type = 'Class' and ./target/model/@type = 'Enumeration')]"
```
Comment (costezki) [inline src/reasoning-layer-lib/connectors-reasoning-layer.xsl:108]: nice filter

File: src/reasoning-layer-lib/connectors-reasoning-layer.xsl [hunk 3]
```diff
@@ -386,37 +392,41 @@
         <xsl:variable name="connectorDirection" select="$connector/properties/@direction"/>
         <xsl:variable name="datatypeURI"
             select="f:buildURIfromLexicalQName('xsd:integer', fn:false())"/>
+<!--        this is first restriction content-->
```
Comment (costezki) [inline src/reasoning-layer-lib/connectors-reasoning-layer.xsl:395]: would be nice to find a way to refactor and optimise this in the future

File: src/reasoning-layer-lib/elements-reasoning-layer.xsl [hunk 4]
```diff
@@ -133,9 +133,9 @@
         <xsl:choose>
             <xsl:when
                 test="fn:count($attributesWithSameName) = 1 or fn:count($distinctAttributeTypesFound) = 1">
-                <xsl:variable name="className" select="$attributesWithSameName/../../@name"/>
+<!--                <xsl:variable name="className" select="$attributesWithSameName/../../@name"/>
```
Comment (costezki) [inline src/reasoning-layer-lib/elements-reasoning-layer.xsl]: can these lines be deleted?

File: test/unitTests/test-reasoning-layer-lib/test-connectors-reasoning-layer.xspec [hunk 5]
```diff
@@ -199,5 +199,41 @@
         
     </x:scenario>
     
+    <x:scenario label="Scenario for connector range -> epo:hasAwardStatus">
```
Comment (costezki) [inline test/unitTests/test-reasoning-layer-lib/test-connectors-reasoning-layer.xspec:202]: very well done. these new tests will assure us against regressions on our code.

File: test/unitTests/test-shacl-shape-lib/test-connectors-shacl-shape.xspec [hunk 6]
```diff
@@ -43,7 +43,20 @@
         <x:expect label="there is a sh:path" test="boolean(sh:NodeShape/sh:property/sh:PropertyShape/sh:path)"/>
         <x:expect label="there is a sh:name" test="boolean(sh:NodeShape/sh:property/sh:PropertyShape/sh:name)"/>
         <x:expect label="there is a sh:minCount" test="boolean(sh:NodeShape/sh:property/sh:PropertyShape/sh:minCount)"/>
-        <x:expect label="there is no sh:maxCount" test="not(boolean(sh:NodeShape/sh:property/sh:PropertyShape/sh:maxCount))"/>
+        <x:expect label="there is no sh:maxCount" test="boolean(sh:NodeShape/sh:property/sh:PropertyShape/sh:maxCount)"/>
```
Comment (costezki) [inline test/unitTests/test-shacl-shape-lib/test-connectors-shacl-shape.xspec:46]: nice test

**Review by costezki — APPROVED** [review]
well done

File: src/reasoning-layer-lib/elements-reasoning-layer.xsl (hunk 4 above)
Comment (Dragos0000) [reply src/reasoning-layer-lib/elements-reasoning-layer.xsl]: sure

File: src/reasoning-layer-lib/connectors-reasoning-layer.xsl (hunk 3 above)
Comment (Dragos0000) [reply src/reasoning-layer-lib/connectors-reasoning-layer.xsl:395]: created a new ticket in the backlog

**Pull request merged**